### DISCUSSION TITLE
[Bug] Fine Tune YourTTS with Around 100 Audio Samples

### DISCUSSION THREAD

**Discussion by ZhichaoWang970201** [coqui-ai/TTS#2708] [General Q&A] [answered]
### Describe the bug

@Edresson I want to fine tun yourTTS with around 100 audio examples. However, the current results give me a not so good results. I have attached my train_yourtts file.

Could you please provide me some suggestions? Thank you.

### To Reproduce

```
import os

import torch
from trainer import Trainer, TrainerArgs

from TTS.bin.compute_embeddings import compute_embeddings
from TTS.bin.resample import resample_files
from TTS.config.shared_configs import BaseDatasetConfig
from TTS.tts.configs.vits_config import VitsConfig
from TTS.tts.datasets import load_tts_samples
from TTS.tts.models.vits import CharactersConfig, Vits, VitsArgs, VitsAudioConfig
from TTS.utils.downloaders import download_vctk

torch.set_num_threads(24)

# pylint: disable=W0105
"""
    This recipe replicates the first experiment proposed in the YourTTS paper (https://arxiv.org/abs/2112.02418).
    YourTTS model is based on the VITS model however it uses external speaker embeddings extracted from a pre-trained speaker encoder and has small architecture changes.
    In addition, YourTTS can be trained in multilingual data, however, this recipe replicates the single language training using the VCTK dataset.
    If you are interested in multilingual training, we have commented on parameters on the VitsArgs class instance that should be enabled for multilingual training.
    In addition, you will need to add the extra datasets following the VCTK as an example.
"""
CURRENT_PATH = os.path.dirname(os.path.abspath(__file__))

# Name of the run for the Trainer
RUN_NAME = "YourTTS-EN-VCTK-FT"

# Path where you want to save the models outputs (configs, checkpoints and tensorboard logs)
OUT_PATH = os.path.join(os.path.dirname(os.path.abspath(__file__)), "YourTTS_FT")  # "/raid/coqui/Checkpoints/original-YourTTS/"

# If you want to do transfer learning and speedup your training you can set here the path to the original YourTTS model
RESTORE_PATH = "/root/.local/share/tts/tts_models--multilingual--multi-dataset--your_tts/model_file.pth"  # "/root/.local/share/tts/tts_models--multilingual--multi-dataset--your_tts/model_file.pth"

# This paramter is useful to debug, it skips the training epochs and just do the evaluation  and produce the test sentences
SKIP_TRAIN_EPOCH = False

# Set here the batch size to be used in training and evaluation
BATCH_SIZE = 32

# Training Sampling rate and the target sampling rate for resampling the downloaded dataset (Note: If you change this you might need to redownload the dataset !!)
# Note: If you add new datasets, please make sure that the dataset sampling rate and this parameter are matching, otherwise resample your audios
SAMPLE_RATE = 16000

# Max audio length in seconds to be used in training (every audio bigger than it will be ignored)
MAX_AUDIO_LEN_IN_SECONDS = 10

### Download VCTK dataset
VCTK_DOWNLOAD_PATH = os.path.join(CURRENT_PATH, "VCTK")
# Define the number of threads used during the audio resampling
#NUM_RESAMPLE_THREADS = 10
# Check if VCTK dataset is not already downloaded, if not download it
#if not os.path.exists(VCTK_DOWNLOAD_PATH):
    #print(">>> Downloading VCTK dataset:")
    #download_vctk(VCTK_DOWNLOAD_PATH)
    #resample_files(VCTK_DOWNLOAD_PATH, SAMPLE_RATE, file_ext="flac", n_jobs=NUM_RESAMPLE_THREADS)

# init configs
# dataset config for one of the pre-defined datasets
vctk_config = BaseDatasetConfig(
        formatter="ljspeech", meta_file_train="metadata.txt", language="en-us", path="./MyTTSDataset")

#vctk_config = BaseDatasetConfig(
#    formatter="vctk",
#    dataset_name="vctk",
#    meta_file_train="",
#    meta_file_val="",
#    path=VCTK_DOWNLOAD_PATH,
#    language="en",
#    ignored_speakers=[
#        "p261",
#        "p225",
#        "p294",
#        "p347",
#        "p238",
#        "p234",
#        "p248",
#        "p335",
#        "p245",
#        "p326",
#        "p302",
#    ],  # Ignore the test speakers to full replicate the paper experiment
#)

# Add here all datasets configs, in our case we just want to train with the VCTK dataset then we need to add just VCTK. Note: If you want to add new datasets, just add them here and it will automatically compute the speaker embeddings (d-vectors) for this new dataset :)
DATASETS_CONFIG_LIST = [vctk_config]

### Extract speaker embeddings
SPEAKER_ENCODER_CHECKPOINT_PATH = (
    "https://github.com/coqui-ai/TTS/releases/download/speaker_encoder_model/model_se.pth.tar"
)
SPEAKER_ENCODER_CONFIG_PATH = "https://github.com/coqui-ai/TTS/releases/download/speaker_encoder_model/config_se.json"

D_VECTOR_FILES = []  # List of speaker embeddings/d-vectors to be used during the training

# Iterates all the dataset configs checking if the speakers embeddings are already computated, if not compute it
for dataset_conf in DATASETS_CONFIG_LIST:
    # Check if the embeddings weren't already computed, if not compute it
    embeddings_file = os.path.join(dataset_conf.path, "speakers.pth")
    if not os.path.isfile(embeddings_file):
        print(f">>> Computing the speaker embeddings for the {dataset_conf.dataset_name} dataset")
        compute_embeddings(
            SPEAKER_ENCODER_CHECKPOINT_PATH,
            SPEAKER_ENCODER_CONFIG_PATH,
            embeddings_file,
            old_spakers_file=None,
            config_dataset_path=None,
            formatter_name=dataset_conf.formatter,
            dataset_name=dataset_conf.dataset_name,
            dataset_path=dataset_conf.path,
            meta_file_train=dataset_conf.meta_file_train,
            meta_file_val=dataset_conf.meta_file_val,
            disable_cuda=False,
            no_eval=False,
        )
    D_VECTOR_FILES.append(embeddings_file)


# Audio config used in training.
audio_config = VitsAudioConfig(
    sample_rate=SAMPLE_RATE,
    hop_length=256,
    win_length=1024,
    fft_size=1024,
    mel_fmin=0.0,
    mel_fmax=None,
    num_mels=80,
)

# Init VITSArgs setting the arguments that are needed for the YourTTS model
model_args = VitsArgs(
    d_vector_file=D_VECTOR_FILES,
    use_d_vector_file=True,
    d_vector_dim=512,
    num_layers_text_encoder=10,
    speaker_encoder_model_path=SPEAKER_ENCODER_CHECKPOINT_PATH,
    speaker_encoder_config_path=SPEAKER_ENCODER_CONFIG_PATH,
    resblock_type_decoder="2",  # In the paper, we accidentally trained the YourTTS using ResNet blocks type 2, if you like you can use the ResNet blocks type 1 like the VITS model
    # Useful parameters to enable the Speaker Consistency Loss (SCL) described in the paper
    # use_speaker_encoder_as_loss=True,
    # Useful parameters to enable multilingual training
    use_language_embedding=True,
    embedded_language_dim=4,
)

# General training config, here you can change the batch size and others useful parameters
config = VitsConfig(
    lr=0.00001,
    output_path=OUT_PATH,
    model_args=model_args,
    run_name=RUN_NAME,
    project_name="YourTTS",
    run_description="""
            - Original YourTTS trained using VCTK dataset
        """,
    dashboard_logger="tensorboard",
    logger_uri=None,
    audio=audio_config,
    batch_size=BATCH_SIZE,
    batch_group_size=48,
    eval_batch_size=BATCH_SIZE,
    num_loader_workers=8,
    eval_split_max_size=256,
    print_step=50,
    plot_step=100,
    log_model_step=1000,
    save_step=5000,
    save_n_checkpoints=2,
    save_checkpoints=True,
    target_loss="loss_1",
    print_eval=False,
    use_phonemes=False,
    phonemizer="espeak",
    phoneme_language="en",
    compute_input_seq_cache=True,
    add_blank=True,
    text_cleaner="multilingual_cleaners",
    characters=CharactersConfig(
        characters_class="TTS.tts.models.vits.VitsCharacters",
        pad="_",
        eos="&",
        bos="*",
        blank=None,
        characters="ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz\u00af\u00b7\u00df\u00e0\u00e1\u00e2\u00e3\u00e4\u00e6\u00e7\u00e8\u00e9\u00ea\u00eb\u00ec\u00ed\u00ee\u00ef\u00f1\u00f2\u00f3\u00f4\u00f5\u00f6\u00f9\u00fa\u00fb\u00fc\u00ff\u0101\u0105\u0107\u0113\u0119\u011b\u012b\u0131\u0142\u0144\u014d\u0151\u0153\u015b\u016b\u0171\u017a\u017c\u01ce\u01d0\u01d2\u01d4\u0430\u0431\u0432\u0433\u0434\u0435\u0436\u0437\u0438\u0439\u043a\u043b\u043c\u043d\u043e\u043f\u0440\u0441\u0442\u0443\u0444\u0445\u0446\u0447\u0448\u0449\u044a\u044b\u044c\u044d\u044e\u044f\u0451\u0454\u0456\u0457\u0491\u2013!'(),-.:;? ",
        punctuations="!'(),-.:;? ",
        phonemes="",
        is_unique=True,
        is_sorted=True,
    ),
    phoneme_cache_path=None,
    precompute_num_workers=12,
    start_by_longest=True,
    datasets=DATASETS_CONFIG_LIST,
    cudnn_benchmark=False,
    max_audio_len=SAMPLE_RATE * MAX_AUDIO_LEN_IN_SECONDS,
    mixed_precision=False,
    test_sentences=[
        [
            "It took me quite a long time to develop a voice, and now that I have it I'm not going to be silent.",
            #"VCTK_p277",
            "ljspeech",
            None,
            "en-us",
        ],
        [
            "Be a voice, not an echo.",
            #"VCTK_p239",
            "ljspeech",
            None,
            "en-us",
        ],
        [
            "I'm sorry Dave. I'm afraid I can't do that.",
            #"VCTK_p258",
            "ljspeech",
            None,
            "en-us",
        ],
        [
            "This cake is great. It's so delicious and moist.",
            #"VCTK_p244",
            "ljspeech",
            None,
            "en-us",
        ],
        [
            "Prior to November 22, 1963.",
            #"VCTK_p305",
            "ljspeech",
            None,
            "en-us",
        ],
    ],

    # Enable the weighted sampler
    use_weighted_sampler=True,
    # Ensures that all speakers are seen in the training batch equally no matter how many samples each speaker has
    weighted_sampler_attrs={"speaker_name": 1.0},
    weighted_sampler_multipliers={},
    # It defines the Speaker Consistency Loss (SCL) α to 9 like the paper
    speaker_encoder_loss_alpha=9.0,
)
# Load all the datasets samples and split traning and evaluation sets
#train_samples, eval_samples = load_tts_samples(
#    config.datasets,
#    eval_split=True,
#    eval_split_max_size=config.eval_split_max_size,
#    eval_split_size=config.eval_split_size,
#)
train_samples, eval_samples = load_tts_samples(
    config.datasets,
    eval_split=True,
    #eval_split_max_size=config.eval_split_max_size,
    #eval_split_size=config.eval_split_size,
    eval_split_size=32,
)

# Init the model
model = Vits.init_from_config(config)

# Init the trainer and 🚀
trainer = Trainer(
    TrainerArgs(restore_path=RESTORE_PATH, skip_train_epoch=SKIP_TRAIN_EPOCH),
    config,
    output_path=OUT_PATH,
    model=model,
    train_samples=train_samples,
    eval_samples=eval_samples,
)

trainer.fit()
```

### Expected behavior

Better quality on TTS for new identity.

### Environment

```shell
{
    "CUDA": {
        "GPU": [
            "NVIDIA A10G"
        ],
        "available": true,
        "version": "11.7"
    },
    "Packages": {
        "PyTorch_debug": false,
        "PyTorch_version": "1.13.1",
        "TTS": "0.14.3",
        "numpy": "1.21.6"
    },
    "System": {
        "OS": "Linux",
        "architecture": [
            "64bit",
            ""
        ],
        "processor": "x86_64",
        "python": "3.7.0",
        "version": "#40~20.04.1-Ubuntu SMP Mon Apr 24 00:21:13 UTC 2023"
    }
}
```

**Comment by Edresson** [comment]
Hi @ZhichaoWang970201, 

I recommend you follow the same approach that we did on [YourTTS paper](https://arxiv.org/abs/2112.02418). You need to train the model with a training dataset plus your speaker data (like VCTK + your data). Also, you need to enable the speaker-weighted sampler and guarantee that your speaker samples appear at least 1/4 of the time in the batch. For example, assume that your speaker name is "my_custom_speaker" and that you are using VCTK data to train. Given that VCTK has 108 speakers, you need to set the weight multiplier for your speaker ("my_custom_speaker") to 27 (108/4) to ensure that the samples from your speaker appear in 1/4 of the batch. 
```
use_weighted_sampler=True,
weighted_sampler_attrs={"speaker_name": 1.0},
weighted_sampler_multipliers={"my_custom_speaker":27},
```

Note that for adding a new dataset you need to just add the new config on the `DATASETS_CONFIG_LIST`. Example:
```
DATASETS_CONFIG_LIST = [vctk_config, my_custom_dataset_config]
```

**Comment by ZhichaoWang970201** [comment]
@Edresson Thank you so much for your help. Now the generated voice sounds more like the speaker. Can I ask several more questions?
1. suppose my speaker is named p400 using VCTK formatter, should the name be "p400" or "VCTK_p400". Both methods do not return me an error.
2. I think the generated audio quality is lower. The connection between is blurry. Like buzzing.
3. In YourTTS paper, an initial learning rate of 0.0002 decaying exponentially by a gamma of 0.999875 [41]. Do you use the same learning rate in the fine-tuning process or do your utilized a smaller learning rate?
4. The text include "AI" and "AWS", which should pronounce each character. The model seems to be not good at pronouncing such cases. Is there a way to solve this problem?
Thank you in advance.

**Comment by Edresson** [comment, marked as answer]
> @Edresson Thank you so much for your help. Now the generated voice sounds more like the speaker. Can I ask several more questions?
> 
> 1. suppose my speaker is named p400 using VCTK formatter, should the name be "p400" or "VCTK_p400". Both methods do not return me an error.
> 2. I think the generated audio quality is lower. The connection between is blurry. Like buzzing.
> 3. In YourTTS paper, an initial learning rate of 0.0002 decaying exponentially by a gamma of 0.999875 [41]. Do you use the same learning rate in the fine-tuning process or do your utilized a smaller learning rate?
> 4. The text include "AI" and "AWS", which should pronounce each character. The model seems to be not good at pronouncing such cases. Is there a way to solve this problem?
>    Thank you in advance.


1. You need to use the speaker name returned by the formatter. The VCTK formatter added "VCTK_" to the speaker name. Given that you need to use "VCTK_p400".
2. I am not sure what you are comparing, but in general, you might need more training. You are restoring the model from the checkpoint "/root/.local/share/tts/tts_models--multilingual--multi-dataset--your_tts/model_file.pth" This checkpoint dont have the discriminator weights. In this way the discriminator will give bad scores and then until the audio start to be good again will take a while. You can get the full checkpoint and speedup the convergence here: https://github.com/Edresson/YourTTS#checkpoints

3. In yourTTS paper, I kept the same learning rate. However, I fully restored the model including the optimizer. Given that the learning rate was small during the fine-tuning.
4. You can try to train the model with phonemes. You can use speak phonemizer easily. It also might solve pronunciation issues.

**Comment by ZhichaoWang970201** [comment]
@Edresson I am sorry that I still have problems.
I try to use your pretrained checkpoint in the following link: https://drive.google.com/drive/folders/10hX5B_h0dzroY7bVNPodC8hzhz4nklzR
I modify the config as the following:
{
    "model": "vits",
    "run_name": "YourTTS-EN-VCTK-FT",
    "run_description": "",
    "epochs": 1000,
    "batch_size": 32,
    "eval_batch_size": 32,
    "mixed_precision": false,
    "scheduler_after_epoch": true,
    "run_eval": true,
    "test_delay_epochs": 0,
    "print_eval": false,
    "dashboard_logger": "tensorboard",
    "print_step": 50,
    "plot_step": 100,
    "model_param_stats": false,
    "project_name": "YourTTS",
    "log_model_step": 1000,
    "wandb_entity": null,
    "save_step": 5000,
    "checkpoint": true,
    "keep_all_best": false,
    "keep_after": 10000,
    "num_loader_workers": 4,
    "num_eval_loader_workers": 4,
    "use_noise_augment": false,
    "use_language_weighted_sampler": false,
    "output_path": "/home/ubuntu/zhichao/TTS/YourTTS_FT",
    "distributed_backend": "nccl",
    "distributed_url": "tcp://localhost:54321",
    "audio": {
        "fft_size": 1024,
        "win_length": 1024,
        "hop_length": 256,
        "frame_shift_ms": null,
        "frame_length_ms": null,
        "stft_pad_mode": "reflect",
        "sample_rate": 16000,
        "resample": false,
        "preemphasis": 0.0,
        "ref_level_db": 20,
        "do_sound_norm": false,
        "log_func": "np.log",
        "do_trim_silence": true,
        "trim_db": 45,
        "power": 1.5,
        "griffin_lim_iters": 60,
        "num_mels": 80,
        "mel_fmin": 0.0,
        "mel_fmax": null,
        "spec_gain": 1,
        "do_amp_to_db_linear": false,
        "do_amp_to_db_mel": true,
        "signal_norm": false,
        "min_level_db": -100,
        "symmetric_norm": true,
        "max_norm": 4.0,
        "clip_norm": true,
        "stats_path": null
    },
    "use_phonemes": false,
    "use_espeak_phonemes": false,
    "phoneme_language": "pt-br",
    "compute_input_seq_cache": false,
    "text_cleaner": "multilingual_cleaners",
    "enable_eos_bos_chars": false,
    "test_sentences_file": "",
    "phoneme_cache_path": null,
    "characters": {
        "pad": "_",
        "eos": "&",
        "bos": "*",
        "characters": "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz\u00af\u00b7\u00df\u00e0\u00e1\u00e2\u00e3\u00e4\u00e6\u00e7\u00e8\u00e9\u00ea\u00eb\u00ec\u00ed\u00ee\u00ef\u00f1\u00f2\u00f3\u00f4\u00f5\u00f6\u00f9\u00fa\u00fb\u00fc\u00ff\u0101\u0105\u0107\u0113\u0119\u011b\u012b\u0131\u0142\u0144\u014d\u0151\u0153\u015b\u016b\u0171\u017a\u017c\u01ce\u01d0\u01d2\u01d4\u0430\u0431\u0432\u0433\u0434\u0435\u0436\u0437\u0438\u0439\u043a\u043b\u043c\u043d\u043e\u043f\u0440\u0441\u0442\u0443\u0444\u0445\u0446\u0447\u0448\u0449\u044a\u044b\u044c\u044d\u044e\u044f\u0451\u0454\u0456\u0457\u0491\u2013",
        "punctuations": "!'(),-.:;? ",
        "phonemes": "iy\u0268\u0289\u026fu\u026a\u028f\u028ae\u00f8\u0258\u0259\u0275\u0264o\u025b\u0153\u025c\u025e\u028c\u0254\u00e6\u0250a\u0276\u0251\u0252\u1d7b\u0298\u0253\u01c0\u0257\u01c3\u0284\u01c2\u0260\u01c1\u029bpbtd\u0288\u0256c\u025fk\u0261q\u0262\u0294\u0274\u014b\u0272\u0273n\u0271m\u0299r\u0280\u2c71\u027e\u027d\u0278\u03b2fv\u03b8\u00f0sz\u0283\u0292\u0282\u0290\u00e7\u029dx\u0263\u03c7\u0281\u0127\u0295h\u0266\u026c\u026e\u028b\u0279\u027bj\u0270l\u026d\u028e\u029f\u02c8\u02cc\u02d0\u02d1\u028dw\u0265\u029c\u02a2\u02a1\u0255\u0291\u027a\u0267\u025a\u02de\u026b'\u0303' ",
        "unique": true
    },
    "batch_group_size": 0,
    "loss_masking": null,
    "min_seq_len": 60,
    "max_seq_len": 270,
    "compute_f0": false,
    "compute_linear_spec": true,
    "add_blank": true,
    "datasets": [
        {
            "name": "vctk",
            "path": "/home/ubuntu/zhichao/TTS/VCTK",
            "meta_file_train": null,
            "ununsed_speakers": [
                "p225",
                "p234",
                "p238",
                "p245",
                "p248",
                "p261",
                "p294",
                "p302",
                "p326",
                "p335",
                "p347"
            ],
            "language": "en",
            "meta_file_val": null,
            "meta_file_attn_mask": ""
        }
    ],
    "optimizer": "AdamW",
    "optimizer_params": {
        "betas": [
            0.8,
            0.99
        ],
        "eps": 1e-09,
        "weight_decay": 0.01
    },
    "lr_scheduler": "",
    "lr_scheduler_params": null,
    "test_sentences": [
        [
            "It took me quite a long time to develop a voice, and now that I have it I'm not going to be silent.",
            "VCTK_p225",
            null,
            "en"
        ],
        [
            "It took me quite a long time to develop a voice, and now that I have it I'm not going to be silent.",
            "ED",
            null,
            "en"
        ],
        [
            "It took me quite a long time to develop a voice, and now that I have it I'm not going to be silent.",
            "bernard",
            null,
            "en"
        ],
        [
            "This cake is great. It's so delicious and moist.",
            "VCTK_p234",
            null,
            "en"
        ],
        [
            "This cake is great. It's so delicious and moist.",
            "ED",
            null,
            "en"
        ],
        [
            "This cake is great. It's so delicious and moist.",
            "ezwa",
            null,
            "en"
        ],
        [
            "Hoje \u00e9 fundamental encontrar a raz\u00e3o da exist\u00eancia humana.",
            "ED",
            null,
            "pt-br"
        ],
        [
            "Hoje \u00e9 fundamental encontrar a raz\u00e3o da exist\u00eancia humana.",
            "VCTK_p238",
            null,
            "pt-br"
        ],
        [
            "Hoje \u00e9 fundamental encontrar a raz\u00e3o da exist\u00eancia humana.",
            "gilles_g_le_blanc",
            null,
            "pt-br"
        ],
        [
            "Em muitas cidades a popula\u00e7\u00e3o est\u00e1 diminuindo.",
            "ED",
            null,
            "pt-br"
        ],
        [
            "Em muitas cidades a popula\u00e7\u00e3o est\u00e1 diminuindo.",
            "VCTK_p245",
            null,
            "pt-br"
        ],
        [
            "Em muitas cidades a popula\u00e7\u00e3o est\u00e1 diminuindo.",
            "nadine_eckert_boulet",
            null,
            "pt-br"
        ],
        [
            "Il m'a fallu beaucoup de temps pour d\u00e9velopper une voix, et maintenant que je l'ai, je ne vais pas me taire.",
            "VCTK_p245",
            null,
            "fr-fr"
        ],
        [
            "Il m'a fallu beaucoup de temps pour d\u00e9velopper une voix, et maintenant que je l'ai, je ne vais pas me taire.",
            "ED",
            null,
            "fr-fr"
        ],
        [
            "Il m'a fallu beaucoup de temps pour d\u00e9velopper une voix, et maintenant que je l'ai, je ne vais pas me taire.",
            "ezwa",
            null,
            "fr-fr"
        ],
        [
            "Il m'a fallu beaucoup de temps pour d\u00e9velopper une voix, et maintenant que je l'ai, je ne vais pas me taire.",
            "bernard",
            null,
            "fr-fr"
        ],
        [
            "Il m'a fallu beaucoup de temps pour d\u00e9velopper une voix, et maintenant que je l'ai, je ne vais pas me taire.",
            "gilles_g_le_blanc",
            null,
            "fr-fr"
        ],
        [
            "Il m'a fallu beaucoup de temps pour d\u00e9velopper une voix, et maintenant que je l'ai, je ne vais pas me taire.",
            "nadine_eckert_boulet",
            null,
            "fr-fr"
        ],
        [
            "Il m'a fallu beaucoup de temps pour d\u00e9velopper une voix, et maintenant que je l'ai, je ne vais pas me taire.",
            "zeckou",
            null,
            "fr-fr"
        ]
    ],
    "use_speaker_embedding": false,
    "use_d_vector_file": true,
    "d_vector_dim": 512,
    "model_args": {
        "num_chars": 165,
        "out_channels": 513,
        "spec_segment_size": 62,
        "hidden_channels": 192,
        "hidden_channels_ffn_text_encoder": 768,
        "num_heads_text_encoder": 2,
        "num_layers_text_encoder": 10,
        "kernel_size_text_encoder": 3,
        "dropout_p_text_encoder": 0.1,
        "dropout_p_duration_predictor": 0.5,
        "kernel_size_posterior_encoder": 5,
        "dilation_rate_posterior_encoder": 1,
        "num_layers_posterior_encoder": 16,
        "kernel_size_flow": 5,
        "dilation_rate_flow": 1,
        "num_layers_flow": 4,
        "resblock_type_decoder": 2,
        "resblock_kernel_sizes_decoder": [
            3,
            7,
            11
        ],
        "resblock_dilation_sizes_decoder": [
            [
                1,
                3,
                5
            ],
            [
                1,
                3,
                5
            ],
            [
                1,
                3,
                5
            ]
        ],
        "upsample_rates_decoder": [
            8,
            8,
            2,
            2
        ],
        "upsample_initial_channel_decoder": 512,
        "upsample_kernel_sizes_decoder": [
            16,
            16,
            4,
            4
        ],
        "use_sdp": true,
        "noise_scale": 1.0,
        "inference_noise_scale": 0.667,
        "length_scale": 1,
        "noise_scale_dp": 1.0,
        "inference_noise_scale_dp": 0.8,
        "max_inference_len": null,
        "init_discriminator": true,
        "use_spectral_norm_disriminator": false,
        "use_speaker_embedding": false,
        "num_speakers": 97,
        "speakers_file": null,
        "d_vector_file": [
            "/home/ubuntu/zhichao/TTS/VCTK/speakers.pth"
        ],
        "speaker_embedding_channels": 512,
        "use_d_vector_file": true,
        "d_vector_dim": 512,
        "detach_dp_input": true,
        "use_language_embedding": false,
        "embedded_language_dim": 4,
        "num_languages": 0,
        "use_speaker_encoder_as_loss": true,
        "speaker_encoder_config_path": "https://github.com/coqui-ai/TTS/releases/download/speaker_encoder_model/config_se.json",
        "speaker_encoder_model_path": "https://github.com/coqui-ai/TTS/releases/download/speaker_encoder_model/model_se.pth.tar",
        "fine_tuning_mode": 0,
        "freeze_encoder": false,
        "freeze_DP": false,
        "freeze_PE": false,
        "freeze_flow_decoder": false,
        "freeze_waveform_decoder": false
    },
    "grad_clip": [
        5.0,
        5.0
    ],
    "lr_gen": 0.0002,
    "lr_disc": 0.0002,
    "lr_scheduler_gen": "ExponentialLR",
    "lr_scheduler_gen_params": {
        "gamma": 0.999875,
        "last_epoch": -1
    },
    "lr_scheduler_disc": "ExponentialLR",
    "lr_scheduler_disc_params": {
        "gamma": 0.999875,
        "last_epoch": -1
    },
    "kl_loss_alpha": 1.0,
    "disc_loss_alpha": 1.0,
    "gen_loss_alpha": 1.0,
    "feat_loss_alpha": 1.0,
    "mel_loss_alpha": 45.0,
    "dur_loss_alpha": 1.0,
    "speaker_encoder_loss_alpha": 9.0,
    "return_wav": true,
    "r": 1
}
I mee the following error:
Traceback (most recent call last):
  File "/opt/conda/envs/tts/bin/tts", line 8, in <module>
    sys.exit(main())
  File "/opt/conda/envs/tts/lib/python3.7/site-packages/TTS/bin/synthesize.py", line 398, in main
    args.use_cuda,
  File "/opt/conda/envs/tts/lib/python3.7/site-packages/TTS/utils/synthesizer.py", line 91, in __init__
    self._load_tts(tts_checkpoint, tts_config_path, use_cuda)
  File "/opt/conda/envs/tts/lib/python3.7/site-packages/TTS/utils/synthesizer.py", line 190, in _load_tts
    self.tts_model.load_checkpoint(self.tts_config, tts_checkpoint, eval=True)
  File "/opt/conda/envs/tts/lib/python3.7/site-packages/TTS/tts/models/vits.py", line 1721, in load_checkpoint
    self.load_state_dict(state["model"], strict=strict)
  File "/opt/conda/envs/tts/lib/python3.7/site-packages/torch/nn/modules/module.py", line 1672, in load_state_dict
    self.__class__.__name__, "\n\t".join(error_msgs)))
RuntimeError: Error(s) in loading state_dict for Vits:
	size mismatch for text_encoder.emb.weight: copying a param with shape torch.Size([165, 192]) from checkpoint, the shape in current model is torch.Size([155, 192]).

The code to run:tts --text "April was the month where we shifted our company in a serious way to focus on Data plus AI plus CRM. We had the first generative AI offsite in SF where we discussed all the work going across the clouds which led to lot of new ideas and innovation from EinsteinGPT getting to GA to some innovative work on a SFBrain. My favorite yet to come is a personalized LLM! As the world learns the various ways to harness this new power, its important to realize that in this next generation, the company with the best possibility to win is the one that integrates this power deeply into it is ecosystem and has some moat. You might have read this paper, while its probably not as bleak as it says for Google, its equally true that we are one of the best moat in the industry due to the power of our data and the engagement surfaces we have. It is for us to link the three Data plus AI plus CRM together to realize the value of the moat and create an unparalleled ecosystem. I am excited to have Jayesh and his crew along with our EDC or Einstein teams come together to make this possible." --model_path YourTTS_FT/YourTTS/best_model_latest.pth.tar --config_path YourTTS_FT/YourTTS/config.json --speaker_wav video2.wav --out_path test.wav

Thank you so much.

**Reply by LeiJiang1** [reply]
Do you have any meaningful results? I tried same things but got nothing meaningful.

**Comment by ZhichaoWang970201** [comment]
No, but training from scratch is okay.

Zhichao

LeiJiang1 ***@***.***> 于2023年9月8日周五 06:24写道：

**Reply by xyyimian** [reply]
Hi Zhichao, would you mind sharing your training config which could generate reasonable results? Struggled for days, thanks!

**Comment by ZhichaoWang970201** [comment]
Hey Xing,

I do not think the generated results seem to be reasonable. It always
creates noise. In addition, I did this in my internship and I do not
maintain the training config now.

Best wishes,
Zhichao

Yu Xing ***@***.***> 于2023年11月29日周三 18:58写道：
